### PR TITLE
[Docs] Clarify default value for `allow_no_indices`

### PR DESCRIPTION
**Pull request by matriv**
Add default value to each one of the usages of `allow_no_indices`
since it differs between different APIs.

Relates to: #52534

### CODE REVIEW

File: docs/reference/rest-api/common-parms.asciidoc [hunk 1]
```diff
@@ -42,6 +42,7 @@ tag::allow-no-indices[]
 the request does *not* return an error
 if a wildcard expression
 or `_all` value retrieves only missing or closed indices.
+Defaults to `true`.
```
Comment (javanna) [inline docs/reference/rest-api/common-parms.asciidoc]: I'm afraid that the default value here depends on the API, or is it `true` for all APIs?

Comment (matriv) [reply docs/reference/rest-api/common-parms.asciidoc]: I tried some of them like HEAD, _mapping, _forgemerge and they behave the same.

Comment (martijnvg) [reply docs/reference/rest-api/common-parms.asciidoc]: The default is API dependant. For example the put mapping api defaults to false.

Comment (matriv) [reply docs/reference/rest-api/common-parms.asciidoc]: I see, should we just add a sentence that the default value depends on the API instead?

Comment (jrodewig) [reply docs/reference/rest-api/common-parms.asciidoc]: You can add the default value where the definition is included instead:

```
include::{docdir}/rest-api/common-parms.asciidoc[tag=allow-no-indices]
+
Defaults to `true`.
```

Comment (javanna) [reply docs/reference/rest-api/common-parms.asciidoc]: @jrodewig ++

**Comment by matriv** [comment]
Checked all usages and added the default value to the doc references.

I have another question though, if you check this: https://github.com/elastic/elasticsearch/pull/52635/files#diff-2e0eb1b85aa4eba2d58663601327caffL39
I removed the reference since the value of the `allow_no_indices` has not effect.
You get 404 if the the index is not there regardless of the param.

I'm wondering if we should do the same for the index exists api (`HEAD /<index-pattern>`).
What do you think?

File: docs/reference/docs/update-by-query.asciidoc [hunk 2]
```diff
@@ -158,6 +158,8 @@ or omit to search all indices.
 include::{docdir}/rest-api/common-parms.asciidoc[tag=allow-no-indices]
 
 include::{docdir}/rest-api/common-parms.asciidoc[tag=analyzer]
++
+Defaults to `true`.
```
Comment (jrodewig) [inline docs/reference/docs/update-by-query.asciidoc]: I believe this is on the wrong parameter (analyzer).

However, when I tested `allow_no_indices` on the update by query API, it had no effect on whether an error was returned. Perhaps this should be removed?

File: docs/reference/indices/alias-exists.asciidoc [hunk 3]
```diff
@@ -36,8 +36,6 @@ include::{docdir}/rest-api/common-parms.asciidoc[tag=index]
 [[alias-exists-api-query-params]]
 ==== {api-query-parms-title}
 
-include::{docdir}/rest-api/common-parms.asciidoc[tag=allow-no-indices]
-
```
Comment (jrodewig) [inline docs/reference/indices/alias-exists.asciidoc]: 👍

File: docs/reference/docs/update-by-query.asciidoc (hunk 2 above)
Comment (matriv) [reply docs/reference/docs/update-by-query.asciidoc]: `localhost:9200/index-2*/_update_by_query?conflicts=proceed&allow_no_indices=false`:
```
{
    "error": {
        "root_cause": [
            {
                "type": "index_not_found_exception",
                "reason": "no such index [index-2*]",
                "resource.type": "index_or_alias",
                "resource.id": "index-2*",
                "index_uuid": "_na_",
                "index": "index-2*"
            }
        ],
        "type": "index_not_found_exception",
        "reason": "no such index [index-2*]",
        "resource.type": "index_or_alias",
        "resource.id": "index-2*",
        "index_uuid": "_na_",
        "index": "index-2*"
    },
    "status": 404
}
```

`localhost:9200/index-2*/_update_by_query?conflicts=proceed`:
```
{
    "took": 0,
    "timed_out": false,
    "total": 0,
    "updated": 0,
    "deleted": 0,
    "batches": 0,
    "version_conflicts": 0,
    "noops": 0,
    "retries": {
        "bulk": 0,
        "search": 0
    },
    "throttled_millis": 0,
    "requests_per_second": -1.0,
    "throttled_until_millis": 0,
    "failures": []
}
```

Comment (jrodewig) [reply docs/reference/docs/update-by-query.asciidoc]: Sorry about that. I made a mistake in my test setup.

File: docs/reference/indices/alias-exists.asciidoc (hunk 3 above)
Comment (matriv) [reply docs/reference/indices/alias-exists.asciidoc]: I'm not 100% sure. 
The mechanism is different, but the outcome the same (status 200 or 404).
If `allow_no_indices=false` and no index is matched then we get an exception at the index name resolution phase which is translated to 404.
If `allow_no_indices=true` and no index is matched then no aliases are found and this is again translated to 404.

File: docs/reference/docs/update-by-query.asciidoc (hunk 2 above)
Comment (matriv) [reply docs/reference/docs/update-by-query.asciidoc]: No worries at all, I didn't see your edit in time and tried it once more myself.

Comment (jrodewig) [reply docs/reference/docs/update-by-query.asciidoc]: However, we should still move the default value to the allow_no_indices parm.

File: docs/reference/indices/alias-exists.asciidoc (hunk 3 above)
Comment (jrodewig) [reply docs/reference/indices/alias-exists.asciidoc]: I think we should probably remove documentation for this parameter. It's not useful to the end user and the outcome will be the same regardless. Thanks for catching this @matriv.

**Comment by matriv** [comment]
@jrodewig FYI, I added also this: https://github.com/elastic/elasticsearch/pull/52635/commits/9902b34090f79828165c6b6bf61821d723a83e2d#diff-72be49b35e089e3c18cadc2619d1d933R99

**Review by jrodewig — APPROVED** [review]
LGTM. Thanks @matriv!

**Pull request merged**